jsonrpc/eth_endpoint.go

```diff
@@ -111,28 +111,44 @@ func (e *Eth) SendTransaction(arg *txnArgs) (interface{}, error) {
 	return transaction.Hash.String(), nil
 }
 
-// GetTransactionByHash returns a transaction by his hash
+// GetTransactionByHash returns a transaction by its hash.
+// If the transaction is still pending -> return the txn with some fields omitted
+// If the transaction is sealed into a block -> return the whole txn with all fields
 func (e *Eth) GetTransactionByHash(hash types.Hash) (interface{}, error) {
-	blockHash, ok := e.d.store.ReadTxLookup(hash)
-	if !ok {
-		// txn not found
-		return nil, nil
-	}
-	block, ok := e.d.store.GetBlockByHash(blockHash, true)
-	if !ok {
-		// block receipts not found
-		return nil, nil
-	}
-	for idx, txn := range block.Transactions {
-		if txn.Hash == hash {
-			return toTransaction(txn, block, idx), nil
+	// Check the TxPool for the transaction if it's pending
+	pendingTx, pendingFound := e.d.store.GetPendingTx(hash)
```

Comment:
I think you can return immediately after you found tx

```suggestion
	if pendingTx, pendingFound := e.d.store.GetPendingTx(hash); pendingFound {
		return toPendingTransaction(pendingTx), nil
	}
	
```

Reply:
I've refactored the entire endpoint to be more readable 👍 

ed68a97 & e5c4e1d